runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java

```diff
@@ -68,83 +71,107 @@ public ILogData peek(RuntimeLayout runtimeLayout, long globalAddress) {
         log.trace("Read[{}]: chain {}/{}", globalAddress, numUnits, numUnits);
         // In chain replication, we read from the last unit, though we can optimize if we
         // know where the committed tail is.
-        ILogData ret = CFUtils.getUninterruptibly(
-                runtimeLayout
-                        .getLogUnitClient(globalAddress, numUnits - 1)
-                        .read(globalAddress)).getAddresses()
-                .getOrDefault(globalAddress, null);
-        return ret == null || ret.isEmpty() ? null : ret;
+        ILogData peekResult = CFUtils.getUninterruptibly(runtimeLayout
+                .getLogUnitClient(globalAddress, numUnits - 1)
+                .read(globalAddress)).getAddresses().get(globalAddress);
+
+        return peekResult.isEmpty() ? null : peekResult;
     }
 
     /**
-     * {@inheritDoc}
+     * Reads a list of global addresses from the chain of log unit servers.
+     *
+     * - This method optimizes for the time to wait to hole fill in case an empty address
+     * is encountered.
+     * - If the waitForWrite flag is set to true, when an empty address is encountered,
+     * it waits for the hole to be filled. All subsequent empty addresses are hole filled
```

Comment:
> All subsequent empty addresses are hole filled
within the collection, not all subsequent addresses.

Reply:
Done.